fuse_core/src/callback_wrapper.cpp

```diff
@@ -92,6 +92,13 @@ std::shared_ptr<void> CallbackAdapter::take_data()
       cb_wrapper = callback_queue_.front();
       callback_queue_.pop_front();
     }
+    if (!callback_queue_.empty()) {
```

Comment:
Should this be a while loop instead, or does it "chain"?

Reply:
It will chain, and that's exactly the purpose in putting it in "take_data". Triggering the guard condition here will cause the executor to wake immediately the next time it goes to `rcl_wait()`, see that the waitable is ready, and "take" data from it again.

Reply:
Ahh, gotcha